assets/src/edit-story/app/media/utils/useUploadVideoFrame.js

```diff
@@ -34,14 +34,16 @@ function useUploadVideoFrame({ updateMediaElement }) {
   const { uploadFile } = useUploader(false);
   const { storyId } = useConfig();
   const {
-    actions: { updateElementById },
+    actions: { updateVideoElementsByVideoId },
```

Comment:
@wassgha basically update elements update only for current page. But `updateVideoElementsByVideoId` update all the elements with videoid through all pages.

Reply:
Still something that could be optimized / looked at in the future.

There could be one action for current page and one for all pages. Or a flag on `updateElement`. Plenty of ideas.

Reply:
I agree. For now I'm in preference to separate unless there's a very natural way to combine. @wassgha the method  that you describe is definitely interesting. But before that happens, I think we'll eventually have `resource.id` and `updateResource()` to match/update any resource usage. I'm not a big fun of `videoId` since it's pretty strong tie-in to WordPress. But it's ok for now.